nav2_costmap_2d/plugins/obstacle_layer.cpp

```diff
@@ -218,7 +218,7 @@ void ObstacleLayer::onInitialize()
 
       std::shared_ptr<tf2_ros::MessageFilter<sensor_msgs::msg::LaserScan>> filter(
         new tf2_ros::MessageFilter<sensor_msgs::msg::LaserScan>(
-          *sub, *tf_, global_frame_, 50, rclcpp_node_));
+          *sub, *tf_, global_frame_, 50, rclcpp_node_, tf2::durationFromSec(transform_tolerance)));
```

Comment:
You mind adding this to the other message filters? There's another in this file and 1 in AMCL. There's also one in SLAM Toolbox / STVL each if you're still users of those

Reply:
Edit: You were ahead of me :laughing:

Reply:
already done for STVL (as yes, I am still using it), I am checking the others

Reply:
there's 2 of these in obstacle layer; one for PC2 and another for laser scans

Reply:
Can't believe I missed the PC2 one, done now. Troubleshooting this problem really got the best of me.
However, I don't use SLAM toolbox and I am not familiar with its code, so I will let you fix it there if that's fine.